airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -137,3 +140,86 @@ def extract_records(self, response: requests.Response) -> Iterable[Mapping[str,
                             version["property"] = property_name
                             version[self.entity_primary_key] = primary_key
                             yield version | additional_keys
+
+
+class EngagementsHttpRequester(HttpRequester):
+    """
+    Engagements stream uses different endpoints:
+    - Engagements Recent if start_date/state is less than 30 days and API is able to return all records (<10k), or
+    - Engagements All which extracts all records, but supports filter on connector side
+
+    Recent Engagements API:
+    https://legacydocs.hubspot.com/docs/methods/engagements/get-recent-engagements
+
+    Important: This endpoint returns only last 10k most recently updated records in the last 30 days.
+
+    All Engagements API:
+    https://legacydocs.hubspot.com/docs/methods/engagements/get-all-engagements
+
+    Important:
+
+    1. The stream is declared to use one stream slice from start date(default/config/state) to time.now(). It doesn't have step.
```

Comment:
Thank you calling this out, but in addition can you add that because of how we manage `_use_recent_api`, concurrent processing of date windows is incompatible and therefore does not support using a `step`